### PR TITLE
pkg/flash: update flash to use chips package

### PR DESCRIPTION
**Pull request by rminnich**
Check for the chip ID first, before trying to read the SFDP. If it is found, use those values for density and so on.

In any case, always try to read the sfdp. As time goes by, more and more chips will implement it, and it will go from mostly failing to mostly working.

### CODE REVIEW

**Comment by rminnich** [comment]
I know it seems weird to read the SFDP even when we find the chip, but bear with me. 

Most important: as time goes by, more and more chips will have an SFDP, and it makes sense to read it.

Less important: the packages assumes a valid sfdp in many places, and it's actually a lot simpler to make the effort to read it in, get an empty one, and have operations like DWord fail. Short form: a test was doing a nil deref if I don't read in an empty SFDP :)

**Comment by rminnich** [comment]
FYI: wit this change, I can ID a flash part from an RPI4, connected to QSPI MUX. Next step is read and write.

File: pkg/flash/flash_linux.go [hunk 1]
```diff
@@ -53,36 +46,60 @@ func New(spi SPI) (*Flash, error) {
 	}
 
 	var err error
+	var id chips.ID
+	id, err = f.spi.ID()
+	if err != nil {
+		return nil, fmt.Errorf("Can not ID chip: %w", err)
```
Comment (binjip978) [inline pkg/flash/flash_linux.go:52]: errors strings should not be capitalised https://go.dev/wiki/CodeReviewComments#error-strings

Comment (rminnich) [reply pkg/flash/flash_linux.go]: fixed, thanks.

**Review by binjip978 — APPROVED** [review]

**Pull request merged**